app/src/main/java/com/fstyle/structure_android/utils/navigator/Navigator.java

```diff
@@ -52,16 +67,107 @@ public void startActivityForResult(@NonNull Class<? extends Activity> clazz, Bun
+     *
+     * @param fragment new child fragment
+     */
+    public void goNextChildFragment(@IdRes int containerViewId, Fragment fragment,
+            boolean addToBackStack, int animation, String tag) {
+        FragmentTransaction transaction = mFragment.getChildFragmentManager().beginTransaction();
+        setFragmentTransactionAnimation(transaction, animation);
+        if (addToBackStack) transaction.addToBackStack(fragment.getClass().getSimpleName());
+        transaction.replace(containerViewId, fragment, tag);
+        transaction.commitAllowingStateLoss();
+        mFragment.getChildFragmentManager().executePendingTransactions();
+    }
+
+    /**
+     * Always keep 1 fragment in container
+     *
+     * @return true if fragment stack has pop
+     */
+    public boolean goBackChildFragment() {
+        boolean isShowPrevious = mFragment.getChildFragmentManager().getBackStackEntryCount() > 1;
+        if (isShowPrevious) {
+            mFragment.getChildFragmentManager().popBackStackImmediate();
+        }
+        return isShowPrevious;
+    }
+
+    private void setFragmentTransactionAnimation(FragmentTransaction transaction,
+            @NavigateAnim int animation) {
+        switch (animation) {
+            case NavigateAnim.FADED:
+                transaction.setCustomAnimations(android.R.anim.fade_in, android.R.anim.fade_out,
+                        android.R.anim.fade_in, android.R.anim.fade_out);
+                break;
+            case NavigateAnim.RIGHT_LEFT:
+                transaction.setCustomAnimations(R.anim.slide_right_in, R.anim.slide_left_out,
+                        R.anim.slide_left_in, R.anim.slide_right_out);
+                break;
+            case NavigateAnim.BOTTOM_UP:
+                transaction.setCustomAnimations(R.anim.slide_bottom_in, R.anim.slide_top_out,
+                        R.anim.slide_top_in, R.anim.slide_bottom_out);
+                break;
+            case NavigateAnim.NONE:
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void setActivityTransactionAnimation(@ActivityTransition int animation) {
+        switch (animation) {
+            case ActivityTransition.START:
+                mActivity.overridePendingTransition(R.anim.translate_left, R.anim.translate_still);
+                break;
+            case ActivityTransition.FINISH:
+                mActivity.overridePendingTransition(R.anim.translate_still, R.anim.translate_right);
+                break;
+            case ActivityTransition.NONE:
+                break;
+            default:
+                break;
+        }
+    }
+
+    @IntDef({
+            NavigateAnim.RIGHT_LEFT, NavigateAnim.BOTTOM_UP, NavigateAnim.FADED, NavigateAnim.NONE,
+            NavigateAnim.LEFT_RIGHT
+    })
+    @interface NavigateAnim {
+        int NONE = 0;
+        int RIGHT_LEFT = 1;
+        int BOTTOM_UP = 2;
+        int FADED = 3;
+        int LEFT_RIGHT = 4;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.